repose-aggregator/components/services/http-logging-service/http-logging-service-impl/src/main/scala/org/openrepose/core/services/httplogging/HttpLoggingContextMap.scala

```diff
@@ -0,0 +1,98 @@
+import java.time.Duration
+import java.util
+
+import javax.servlet.http.{HttpServletRequest, HttpServletResponse}
+import org.openrepose.commons.utils.http.CommonHttpHeader
+import org.openrepose.commons.utils.logging.TracingHeaderHelper
+
+import scala.collection.JavaConverters._
+import scala.language.implicitConversions
+import scala.util.Try
+
+/**
+  * Converts an [[HttpLoggingContext]] into a [[Map]] for use by the templating
+  * engine.
+  */
+object HttpLoggingContextMap {
+
+  // TODO: Are these the right keys? Use the single-character style keys?
+  private final val Generators: Map[String, HttpLoggingContext => Any] = Map(
+    "inboundRequestProtocol" -> (_.getInboundRequest.getProtocol),
+    "outboundRequestProtocol" -> (_.getOutboundRequest.getProtocol),
+    "inboundResponseProtocol" -> (_.getInboundResponseProtocol),
+    "inboundRequestMethod" -> (_.getInboundRequest.getMethod),
+    "outboundRequestMethod" -> (_.getOutboundRequest.getMethod),
+    "inboundRequestPath" -> (_.getInboundRequest.getRequestURI),
+    "outboundRequestPath" -> (_.getOutboundRequest.getRequestURI),
+    "inboundRequestQueryString" -> (_.getInboundRequest.getQueryString),
+    "outboundRequestQueryString" -> (_.getOutboundRequest.getQueryString),
+    "inboundRequestUrl" -> (_.getInboundRequest.getRequestURL.toString),
+    "outboundRequestUrl" -> (_.getOutboundRequest.getRequestURL.toString),
+    "inboundRequestHeaders" -> (ctx => headerMap(ctx.getInboundRequest)),
+    "outboundRequestHeaders" -> (ctx => headerMap(ctx.getOutboundRequest)),
+    "outboundResponseHeaders" -> (ctx => headerMap(ctx.getOutboundResponse)),
+    "outboundResponseStatusCode" -> (_.getOutboundResponseStatusCode),
+    "outboundResponseReasonPhrase" -> (_.getOutboundResponseReasonPhrase),
+    "outboundResponseContentLength" -> (_.getOutboundResponseContentLength),
+    "outboundResponseBytesWritten" -> (_.getOutboundResponseBytesWritten),
+    "timeRequestReceived" -> (_.getTimeRequestReceived),
+    "timeToHandleRequest" -> (ctx => Duration.between(ctx.getTimeRequestReceived, ctx.getTimeRequestCompleted)),
+    "timeInOriginService" -> (_.getTimeInOriginService),
+    "localIpAddress" -> (_.getInboundRequest.getLocalAddr),
+    "remoteIpAddress" -> (_.getInboundRequest.getRemoteAddr),
+    "remoteHost" -> (_.getInboundRequest.getRemoteHost),
+    "traceId" -> (ctx => TracingHeaderHelper.getTraceGuid(ctx.getOutboundRequest.getHeader(CommonHttpHeader.TRACE_GUID))),
+    "extensions" -> (_.getExtensions)
+  )
+
+  def from(httpLoggingContext: HttpLoggingContext): Map[String, AnyRef] = {
+    // Removes mappings where a value could not be generated.
+    Generators
+      .mapValues(generator => Try(generator(httpLoggingContext)))
+      .mapValues(tryValue => tryValue.map(Option.apply).getOrElse(None))
+      .collect({ case (key, Some(value: AnyRef)) => key -> value })
+  }
+
+  private def headerMap(request: HttpServletRequest): util.Map[String, _ <: util.List[String]] = {
```

Comment:
While this would be a useful method to have on the wrappers, even if we did, we would still have to lowercase the keys here. That would probably look like:
```
request.getHeaderMap.asScala
      .map({ case (name, value) => name.toLowerCase -> value })
      .asJava
```

So while it's a useful thing in general, it probably wouldn't clean this code up much. Still, it would decrease duplicated code which is nice.